controllers/gce/cmd/mode-updater/main.go

```diff
@@ -0,0 +1,375 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+
+	compute "google.golang.org/api/compute/v1"
+)
+
+var (
+	projectID           string
+	regionName          string
+	targetBalancingMode string
+
+	instanceGroupName string
+
+	s         *compute.Service
+	zones     []*compute.Zone
+	igs       map[string]*compute.InstanceGroup
+	instances []*compute.Instance
+)
+
+const (
+	instanceGroupTemp = "k8s-ig--migrate"
+	balancingModeRATE = "RATE"
+	balancingModeUTIL = "UTILIZATION"
+
+	version = 0.1
+)
+
+func main() {
+	fmt.Println("Backend-Service BalancingMode Updater", version)
+	//flag.Usage
+	flag.Parse()
+
+	args := flag.Args()
+	if len(args) != 3 {
+		log.Fatalf("Expected three arguments: project_id region balancing_mode")
+	}
+	projectID, regionName, targetBalancingMode = args[0], args[1], args[2]
+
+	switch targetBalancingMode {
+	case balancingModeRATE, balancingModeUTIL:
+	default:
+		panic(fmt.Errorf("expected either %s or %s, actual: %v", balancingModeRATE, balancingModeUTIL, targetBalancingMode))
+	}
+
+	switch regionName {
+	case "asia-east1", "asia-northeast1", "europe-west1", "us-central1", "us-east1", "us-west1":
+	default:
+		panic(fmt.Errorf("expected a valid GCP region, actual: %v", regionName))
+	}
+
+	igs = make(map[string]*compute.InstanceGroup)
+
+	tokenSource, err := google.DefaultTokenSource(
+		oauth2.NoContext,
+		compute.CloudPlatformScope,
+		compute.ComputeScope)
+	if err != nil {
+		panic(err)
+	}
+
+	client := oauth2.NewClient(oauth2.NoContext, tokenSource)
+	s, err = compute.New(client)
+	if err != nil {
+		panic(err)
+	}
+
+	// Get Zones
+	zoneFilter := fmt.Sprintf("(region eq %s)", createRegionLink(regionName))
+	zoneList, err := s.Zones.List(projectID).Filter(zoneFilter).Do()
+	if err != nil {
+		panic(err)
+	}
+	zones = zoneList.Items
+
+	// Get instance groups
+	for _, z := range zones {
+		igl, err := s.InstanceGroups.List(projectID, z.Name).Do()
+		if err != nil {
+			panic(err)
+		}
+		for _, ig := range igl.Items {
+			if !strings.HasPrefix(ig.Name, "k8s-ig--") {
+				continue
+			}
+
+			if instanceGroupName == "" {
+				instanceGroupName = ig.Name
+			}
+
+			// Note instances
+			r := &compute.InstanceGroupsListInstancesRequest{InstanceState: "ALL"}
+			instList, err := s.InstanceGroups.ListInstances(projectID, getResourceName(ig.Zone, "zones"), ig.Name, r).Do()
+			if err != nil {
+				panic(err)
+			}
+
+			for _, i := range instList.Items {
+				inst, err := s.Instances.Get(projectID, getResourceName(ig.Zone, "zones"), getResourceName(i.Instance, "instances")).Do()
+				if err != nil {
+					panic(err)
+				}
+
+				instances = append(instances, inst)
+			}
+
+			// Note instance group in zone
+			igs[z.Name] = ig
+		}
+	}
+
+	if instanceGroupName == "" {
```

Comment:
if there is more than one cluster up, this will only be the last one.  Is that what you intended?